web/cobrands/fixmystreet/assets.js

```diff
@@ -311,43 +341,63 @@ fixmystreet.add_assets = function(options) {
             asset_layer.fixmystreet.fault_layer = asset_fault_layer;
         }
 
-        // Set up handlers for selecting/unselecting markers and panning/zooming the map
-        var select_feature_control = new OpenLayers.Control.SelectFeature( asset_layer );
-        asset_layer.events.register( 'featureselected', asset_layer, asset_selected);
-        asset_layer.events.register( 'featureunselected', asset_layer, asset_unselected);
-        asset_layer.events.register( 'loadend', asset_layer, layer_loadend);
-        asset_layer.events.register( 'visibilitychanged', asset_layer, layer_visibilitychanged);
-        fixmystreet.map.events.register( 'zoomend', asset_layer, check_zoom_message_visibility);
+        var hover_feature_control, select_feature_control;
+        if (!options.non_interactive) {
+            // Set up handlers for selecting/unselecting markers
+            select_feature_control = new OpenLayers.Control.SelectFeature( asset_layer );
+            asset_layer.events.register( 'featureselected', asset_layer, asset_selected);
+            asset_layer.events.register( 'featureunselected', asset_layer, asset_unselected);
+            // When panning/zooming the map check that this layer is still correctly shown
+            // and any selected marker is preserved
+            asset_layer.events.register( 'loadend', asset_layer, layer_loadend);
+        }
+
+        // Even if an asset layer is marked as non-interactive it can still have
+        // a hover style which we'll need to set up.
+        if (!options.non_interactive || (options.stylemap && options.stylemap.styles.hover)) {
+            // Set up handlers for simply hovering over a street light marker
+            hover_feature_control = new OpenLayers.Control.SelectFeature(
+                asset_layer,
+                {
+                    hover: true,
+                    highlightOnly: true,
+                    renderIntent: 'hover'
+                }
+            );
+            hover_feature_control.events.register('beforefeaturehighlighted', null, function(e) {
+                // Don't let marker go from selected->hover state,
+                // as it causes some mad flickering effect.
+                if (e.feature.renderIntent == 'select') {
+                    return false;
+                }
+            });
+        }
+        if (options.asset_category) {
+            fixmystreet.map.events.register( 'zoomend', asset_layer, check_zoom_message_visibility);
+        }
+        if (!options.always_visible) {
```

Comment:
If you're on an around page (shows always_visible assets), click map, click back, they disappear and never return.